core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1008,7 +1010,21 @@ public void syncPersistDirectory(InodeDirectory dir, JournalContext journalConte
           MkdirsOptions mkdirsOptions =
               MkdirsOptions.defaults().setCreateParent(false).setOwner(dir.getOwner())
                   .setGroup(dir.getGroup()).setMode(new Mode(dir.getMode()));
-          ufs.mkdirs(ufsUri, mkdirsOptions);
+          if (!ufs.mkdirs(ufsUri, mkdirsOptions)) {
+            // Directory already exists. Try loading the status from ufs.
+            try {
+              UfsStatus status = ufs.getStatus(ufsUri);
+              if (status.isFile()) {
+                throw new InvalidPathException(String.format(
+                    "Error persisting directory. A file exists at the UFS location %s.", ufsUri));
+              }
+              dir.setOwner(status.getOwner())
+                  .setGroup(status.getGroup())
+                  .setMode(status.getMode());
+            } catch (FileNotFoundException e) {
+              // Ignore exception if file is not found
```

Comment:
Should you log a warning? Also, should you throw an exception so it is retried?

Reply:
Good catch. Will do.